src/argowrapper/engine/helpers/argo_engine_helper.py

```diff
@@ -51,9 +52,24 @@ def parse_common_details(
     elif workflow_type == "archived_workflow":
         pass
 
+    user_name_str = ""
+    if workflow_details["metadata"].get("labels"):
+        user_name_str = (
+            workflow_details["metadata"].get("labels").get(GEN3_USER_METADATA_LABEL)
+        )
+
+    if user_name_str:
+        if user_name_str.startswith("user-"):
+            user_name_str = user_name_str[5:]
+        # argo engine encode % as -
+        user_name_str = urllib.parse.unquote(user_name_str.replace("-", "%"))
+        # put actual - back after decoding
+        user_name_str = user_name_str.replace("%", "-")
+
     return {
```

Comment:
there is already a function `convert_username_label_to_gen3username` that does the conversion back. It is actually used and tested here: https://github.com/uc-cdis/argo-wrapper/blob/ae204e5658b4c451e51719b7b525fb48e8a7074c/test/test_argo_engine_helper.py#L22-L34

Reply:
The whole thing is tested here: https://github.com/uc-cdis/argo-wrapper/blob/ae204e5658b4c451e51719b7b525fb48e8a7074c/test/test_argo_engine_helper.py#L50-L64

Reply:
moved functions from test file to helper class file.